crates/chain/src/chain_data.rs

```diff
@@ -215,4 +241,64 @@ impl<I: ChainPosition> FullTxOut<I> {
     }
 }
 
+impl<A: Anchor + ConfirmationHeight> FullTxOut<ObservedAs<A>> {
+    /// Whether the `txout` is considered mature.
+    ///
+    /// This is the alternative version of [`is_mature`] which depends on `chain_position` being a
+    /// [`ObservedAs<A>`] where `A` implements [`Anchor`].
+    ///
+    /// [`is_mature`]: Self::is_mature
+    pub fn is_observed_as_mature(&self, tip: u32) -> bool {
+        if !self.is_on_coinbase {
+            return false;
+        }
+
+        let tx_height = match &self.chain_position {
+            ObservedAs::Confirmed(anchor) => anchor.confirmation_height(),
+            ObservedAs::Unconfirmed(_) => {
+                debug_assert!(false, "coinbase tx can never be unconfirmed");
+                return false;
+            }
+        };
+
+        let age = tip.saturating_sub(tx_height);
+        if age + 1 < COINBASE_MATURITY {
+            return false;
+        }
+
+        true
+    }
+
+    /// Whether the utxo is/was/will be spendable with chain `tip`.
+    ///
+    /// Currently this method does not take into account the locktime.
```

Comment:
This method will never take into a account the lock time since you need the descriptor for that.
```suggestion
    /// This method does not take into account the locktime.
```